services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,47 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if(event == NULL) return ERR_CODE_INVALID_ARG;
+
+  if(xQueueSend(thermalMgrQueueHandle, (void *) event, TEMPERATURE_WAIT_TIME) != pdTRUE) {
+    return ERR_CODE_QUEUE_FULL;
+  }
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t event = {.type = THERMAL_OS_HANDLE};
+  BaseType_t xHigherPriorityTaskWoken = pdFALSE;
```

Comment:
You're currently not making use of this variable. If a higher priority task is unblocked, you need to call `taskYIELD_FROM_ISR ` at the end of the ISR. Take a look at https://www.freertos.org/a00119.html